lib/plugin.js

```diff
@@ -179,6 +180,68 @@ const setupProgress = (axios) => {
   axios.defaults.onDownloadProgress = onProgress
 }<% } %>
 
+<% if (options.proxyCookies) { %>
+const proxyCookies = (ctx, axios, response) => {
+  const parseSetCookies = cookies => {
+    return cookies
+      .map(cookie => cookie.split(';')[0])
+      .reduce((obj, cookie) => ({
+        ...obj,
+        ...parseCookies(cookie)
+      }), {})
+  }
+
+  const serializeCookies = cookies => {
+    return Object
+      .entries(cookies)
+      .map(([name, value]) => `${name}=${encodeURIComponent(value)}`)
+      .join('; ')
+  }
+
+  const mergeSetCookies = (oldCookies, newCookies) => {
+    const cookies = new Map()
+
+    const add = setCookie => {
+      const cookie = setCookie.split(';')[0]
+      const name = Object.keys(parseCookies(cookie))[0]
+
+      cookies.set(name, cookie)
+    }
+
+    oldCookies.forEach(add)
+    newCookies.forEach(add)
+
+    return [...cookies.values()]
+  }
+
+  const arrayify = obj => Array.isArray(obj) ? obj : [obj]
+
+  if (response.headers['set-cookie']) {
+    const setCookies = arrayify(response.headers['set-cookie'])
+
+    // Combine the cookies set on axios with the new cookies and serialize them
+    const cookie = serializeCookies({
+      ...parseCookies(axios.defaults.headers.common.cookie),
```

Comment:
This line can fail with `ERROR argument str must be a string`, cause `axios.defaults.headers.common.cookie` can be `undefined`